### DISCUSSION TITLE
AutoTransformer documentation and attributes?

### DISCUSSION THREAD

**Discussion by nlaws-camus** [dss-extensions/dss-extensions#40] [Q&A] [answered]
Hello! I just discovered that the official OpenDSS has an AutoTransformer class. The official documentation is very sparse and I could not find any documentation for AutoTransformers in the OpenDSSDirect.py nor the dss-python documentation. Am I missing something? I also searched for GICTransformer documentation as it appears that the official OpenDSS has the AutoTransformer inherit from the GICTransformer class. My goal is to model single phase, two-winding regulators with an autotransformer, but I have sequence impedances and not the XHL, XHT, XLT parameters. I have seen the 2013 note from the official OpenDSS docs "Modeling Regulators as Autotransformers" - but the methods described therein seem much more complicated then defining an AutoTransformer directly... if I can figure out _how_ to define an AutoTransformer.

**Comment by PMeira** [comment, marked as answer]
> I also searched for GICTransformer documentation as it appears that the official OpenDSS has the AutoTransformer inherit from the GICTransformer class.

Looks like  the docs on the official site at https://opendss.epri.com/AutoTrans.html were accidentally copied from GICTransformer. It doesn't inherit from it, the code was originally derived from the Transformer model.

We have this: https://dss-extensions.org/dss-format/AutoTrans.html   
The single example listed is the one distributed with the official OpenDSS, here's a direct link to the folder: https://sourceforge.net/p/electricdss/code/HEAD/tree/trunk/Version8/Distrib/Examples/AutoTrans/ (`AutoAuto.dss`, as the other files use normal Transformer objects).

> My goal is to model single phase, two-winding regulators with an autotransformer, but I have sequence impedances and not the XHL, XHT, XLT parameters.

You'll see that the list of parameters is very close to the Transformer model. Besides some names, the main difference is the connection:

> For AutoTrans, Winding 1 is always Series and Winding 2 (the Common winding) is always Wye.
> If only 2 windings, no need to specify connections.

There are some known issues with both reports and RegControls coupled to AutoTrans in the official implementation. We made some changes in AltDSS/DSS C-API to work around those (mostly to avoid memory corruption), but there might be some issues still. The model itself should match the official version, so any issues there are reproduced here.

[From the post](https://sourceforge.net/p/electricdss/discussion/861976/thread/0927b437c9/) you also commented on the OpenDSS forum, this doesn't happen in our implementation:

> Watching the "taps" document, I saw that the min and max value were set on 0 (Figure attached). So, I tried to define MinTap and MaxTap in the code, but after running again the circuit there was no change at all.

But about the voltage issue, I wasn't aware:

> it is not possible to set both windings with the same kV value (like the old way to define the autotransformer

A couple of people lightly mentioned issues in the past (~2020), but nothing this specific.

From a quick inspection, there is already a check for a common voltage, so it's probably either not working or it needs to be checked somewhere else too. I'll watch the post to see it there's any news before we work on that here.

**Comment by nlaws-camus** [comment]
There is also [this SourceForge thread](https://sourceforge.net/p/electricdss/discussion/beginners/thread/671612ab96/?limit=25#97df) where Roger seems to recommend not using the AutoTrans class for single phase regulators :/  I'll stick with the transformer + regulator pattern for now but the AutoTrans (in a trustworthy format) will be a great addition.
